teps/0098-workflows.md

```diff
@@ -125,62 +54,112 @@ simplify the Tekton experience. It can provide a single place to declare all the
 * Make managing Tekton configuration less verbose and repetitive.
-* Support end to end CI workflows from source control
 * Make it easier and simpler for end users to get started with a pure Tekton CI system
-* Allow clear separation of concerns between platform developers and app teams
-* Allow for clear separation between run configuration and pipeline definition?
+* Allow clear separation of concerns between platform developers and app teams,
+including making it easier to separate notifications and other infrastructure out of the Pipelines being run
+* Create a Workflows conformance spec
 
 ### Non-Goals
 
-<!--
-What is out of scope for this TEP?  Listing non-goals helps to focus discussion
-and make progress.
--->
 * Non CI/CD pipelines
-* Installation/deployment of other Tekton projects that Workflows needs as pre-requsities
+* Installation/deployment of other Tekton projects that Workflows needs on behalf of the cluster operator
+  * We can explore using the Operator project to simplify installation, but for an initial implementation
+  the cluster operator will just be directed to install the necessary projects
+* In-tree support for all possible SCMs, cloud storage systems, secrets storage systems, etc.
 
 ### Use Cases
 
-<!--
-Describe the concrete improvement specific groups of users will see if the
-Motivations in this doc result in a fix or feature.
+The following example use cases are targeted by the Workflows project, but may be implemented via several milestones
+(i.e. not all of these features need to be supported in an initial implementation). These use cases are examples, not a comprehensive list of features;
+for example, we want to support CI in multiple SCMs, not just Github.
 
-Consider both the user's role (are they a Task author? Catalog Task user?
-Cluster Admin? etc...) and experience (what workflows or actions are enhanced
-if this problem is solved?).
--->
+1. Github-based CI
 
-1. End users can set up an entire end to end CI pipeline (from source back to run results) using only Tekton 
-1. End users can visualize runs grouped by repository
-1. End users can create an end to end CI/CD workflow e.g. no scripts to set up webhooks, GH Apps etc.
+- The end user would like to trigger a CI PipelineRun whenever a pull request is opened or a commit is pushed to the pull request,
+without setting up any Github webhooks or Apps.
+- If a new commit is pushed to the pull request, any CI PipelineRuns currently running for that branch should be canceled and
+replaced by a new one.
+- They may also want to trigger PipelineRuns based on pull request comments or labels.
+For example, a test could be re-run if a user comments "/retest".
+- They may want to run certain checks only if certain files have changed.
+For example, they may want to skip unit tests if only markdown files are modified.
+- The result of the CI PipelineRun is posted to Github Checks by Tekton, not by a finally Task.
```

Comment:
NIT: maybe "not by a Task owned by the user" - it could be a finally task, and async task, a custom task, or else, but I think the main point here is that it is not controlled by the user, it's provided by workflows.